### DISCUSSION TITLE
Annotation based Indexes only created when using deprecated ensureIndexes() in combination with getEntityModel()

### DISCUSSION THREAD

**Discussion by n-r-a-s** [MorphiaOrg/morphia#2633] [Q&A]
I am struggling with a funny phenomenon in regard to creating indexes based on Annotation @Indexes.

Only if I add the following two statements before an insert (save), the Indexes will be created in MongoDb as expected.
The funny thing is, it realy needs both of the statements in combination.

  ```
  ds.getMapper().getEntityModel(Employee.class);
  ds.ensureIndexes();

  ds.save(....)
```

My Index Annotation:

```
@Entity("employees")
@Indexes({
    @Index(
            fields = @Field("name"),
            options = @IndexOptions(name = "index_name", unique = true)
    )
})
public class Employee {
```
I am using MongoDb Server 7.0
My pom dependencies:

 ```
<dependency>
            <groupId>org.mongodb</groupId>
            <artifactId>mongodb-driver-sync</artifactId>
            <version>4.10.2</version>
</dependency>
<dependency>
            <groupId>dev.morphia.morphia</groupId>
            <artifactId>morphia-core</artifactId>
            <version>2.4.6</version>
</dependency>
```

I was spending quite a time in different forums. Since ensureIndexes is deprecated I found alot of forums dealing with older versions of morphia. Neither in the official documentation nor in other sources I found a hint helping me with my issue.
It just says: no longer use ensureIndexes, use createIndex. Ok, but as far as I understood, then I have to define my Indexes hardcoded instead of the Annotation way. I realy like to use the Annotations. This is one of the reasons why I decided to use Morphia.
Does any one have a hint for me?

**Comment by evanchooly** [comment]
That's the (mostly) correct way to do that.  In 3.0, the application of indexes will be done transparently and will be controlled via either the configuration file or or `MorphiaConfig` instance passed to the `Datastore`.  There's a pending change for 2.4.7 that will update the API.

**Reply by n-r-a-s** [reply]
Thanks alot, this helped! Looking forward to the next version...
